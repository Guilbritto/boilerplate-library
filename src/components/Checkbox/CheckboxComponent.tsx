import React, { useEffect, useState } from 'react'
import { Check } from '@svg/Check'
import { useTheme } from '../../hooks/ThemeContext'
import { Container, EffectArea } from './CheckboxComponent.styles'
import { CheckboxProps } from './CheckboxComponent.types'

const CheckboxComponent = ({ disabled, ...rest }: CheckboxProps) => {
    const theme = useTheme()
    const [isChecked, setIsChecked] = useState(false)
    const [mouseDown, setMouseDown] = useState(false)
    const [mouseUp, setMouseUp] = useState(false)

    const handleMouseUp = (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
        setMouseDown(false)
        setMouseUp(true)
    }
    const handleMouseDown = (
        e: React.MouseEvent<HTMLDivElement, MouseEvent>
    ) => {
        setMouseDown(true)
        setMouseUp(false)
    }
    useEffect(() => {
        if (rest.checked) {
            setIsChecked(true)
        } else {
            setIsChecked(false)
        }
    }, [])

    const handleClick = () => {
        if (!disabled) {
            setIsChecked(!isChecked)
        }
    }

    return (
        <EffectArea theme={theme} disabled={disabled}>
            <Container
                theme={theme}
                disabled={disabled}
                onClick={handleClick}
                isChecked={isChecked}
            >
                <div>
                    <input
                        type="checkbox"
                        checked={isChecked}
                        disabled={disabled}
                        {...rest}
                    />
                    {isChecked && !disabled && Check}
                </div>
            </Container>
        </EffectArea>
    )
}

export default CheckboxComponent
