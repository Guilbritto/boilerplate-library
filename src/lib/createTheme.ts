import { defaultTheme } from "../styles/defaultTheme";

export const createTheme = (theme: typeof defaultTheme) => {
  return {
    ...defaultTheme,
    colors: {
      ...defaultTheme.colors,
      ...theme.colors,
    },
    fontSizes: {
      ...defaultTheme.fontSizes,
      ...theme.fontSizes,
    },
  } as typeof defaultTheme;
}