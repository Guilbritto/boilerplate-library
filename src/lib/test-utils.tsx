import { render, RenderOptions } from '@testing-library/react'
import React, { ReactElement, FC } from 'react'
import { ThemeProvider } from 'styled-components'
import { defaultTheme } from '../styles/defaultTheme'

const Providers: FC = ({ children }) => {
    return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
}

/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
