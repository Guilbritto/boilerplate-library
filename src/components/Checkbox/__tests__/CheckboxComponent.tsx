import { render, screen } from '../../../lib/test-utils'

import CheckboxComponent from '../CheckboxComponent'

describe('<CheckboxComponent />', () => {
    it('should render the component', () => {
        const { container } = render(<CheckboxComponent />)

        expect(container.firstChild).toMatchSnapshot()
    })
})
