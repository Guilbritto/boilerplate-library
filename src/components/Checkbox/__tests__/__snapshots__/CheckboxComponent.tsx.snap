// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxComponent /> should render the component 1`] = `
<div
  class="sc-gsTEea dzWFCL"
>
  <div
    class="sc-bdfBQB jZXqOZ"
  >
    <div>
      <input
        type="checkbox"
      />
    </div>
  </div>
</div>
`;
